### DISCUSSION TITLE
useLoaderData wont revalidate after get submit

### DISCUSSION THREAD

**Discussion by feledori** [remix-run/remix#3062] [Q&A] [answered]
I currently have a diagram setup to show data. On inital pageload it recieves data from 1 month in the past to current data and I have a datepicker setup to change the daterange. On change of the date range i use submit from useSubmit to make a new get request with the dates appended as params. In the loader function I get the dates and fetch the new data. I don't know why but useLoaderData wont revalidate my data now. It used to work thats the strange thing about it. I tried using useFetcher and even navigate but the data wont revalidate and state wont update.

The loader data loads the new data, everything from the data fetching side is working fine but it wont update...

Any ideas why? Is this intended?

**Comment by kiliman** [comment, marked as answer]
Here's an example showing how `useSubmit` is re-rendered with updated data. Let me know if you have any questions.

https://codesandbox.io/s/remix-loader-submit-31248e?file=/app/routes/data.tsx

**Reply by feledori** [reply]
I just found the bug and it had nothing to do with Remix... Sorry to waste your time, but thanks for taking the time to answer!

**Reply by sergiodxa** [reply]
@felixmayr it could be useful if you shared what caused the issue so if someone finds this discussion in the future it could see some possible fix

**Reply by feledori** [reply]
@sergiodxa It was one of my components which memoized its data and I didn't properly configure the dependency array. It was quite deeply nested so I didn't catch the bug.
